test/unit/workDiv/src/WorkDivForKernelTest.cpp

```diff
@@ -0,0 +1,182 @@
+                     Acc>)
+    {
+        // Expected valid workdiv for the specific kernel
+        CHECK(workDiv == WorkDiv{Vec{2}, Vec{512}, Vec{1}});
+    }
+    // ApiHipRt
+    else if constexpr(std::is_same_v<
+                          alpaka::AccGpuUniformCudaHipRt<alpaka::ApiHipRt, alpaka::Dim<Acc>, alpaka::Idx<Acc>>,
+                          Acc>)
+    {
+        // Expected valid workdiv for this specific kernel
+        CHECK(workDiv == WorkDiv{Vec{1}, Vec{1024}, Vec{1}});
+    }
+#endif
+
+#if !defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && !defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    {
+        // CPU must have only 1 thread per block. In other words, number of blocks is equal to number of threads.
+        CHECK(workDiv == WorkDiv{Vec{1024}, Vec{1}, Vec{1}});
+        // Test a new 1D workdiv. Threads per block can not be larger than 1 for CPU. Hence 2 is not valid.
+        auto const& workDiv1DUsingInitList = alpaka::WorkDivMembers{Vec{1024}, Vec{2}, Vec{1}};
+        auto const isWorkDivValidForCPUKernel
+            = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, workDiv1DUsingInitList);
+        CHECK(isWorkDivValidForCPUKernel == false);
+    }
+#endif
+}
+
+using TestAccs2D = alpaka::test::EnabledAccs<alpaka::DimInt<2u>, std::uint32_t>;
+
+TEMPLATE_LIST_TEST_CASE("getFunctionAttributes.2D.withIdx", "[workDivKernel]", TestAccs2D)
+{
+    using Acc = TestType;
+    using Idx = alpaka::Idx<Acc>;
+    using Dim = alpaka::Dim<Acc>;
+    using Vec = alpaka::Vec<Dim, Idx>;
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+    auto const platform = alpaka::Platform<Acc>{};
+    auto const dev = alpaka::getDevByIdx(platform, 0);
+
+    TestKernelWithManyRegisters kernel;
+    size_t const kernelArg = 200ul;
+    auto const& bundeledKernel = alpaka::makeKernelBundle<Acc>(kernel, kernelArg);
+
+    // Test getValidWorkDivForKernel function for 1024x2 threads per grid.
+    auto const workDiv = alpaka::getValidWorkDivForKernel(dev, bundeledKernel, Vec{1024, 2}, Vec{1, 1});
+    // Test isValidWorkDivKernel function
+    auto const isWorkDivValidForKernel = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, workDiv);
+    CHECK(isWorkDivValidForKernel == true);
+
+// If Acc is Cuda or Hip
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    // For ApiCudaRt
+    if constexpr(std::is_same_v<
+                     alpaka::AccGpuUniformCudaHipRt<alpaka::ApiCudaRt, alpaka::Dim<Acc>, alpaka::Idx<Acc>>,
+                     Acc>)
+    {
+        // Expected valid workdiv for this kernel.
+        CHECK(workDiv == WorkDiv{Vec{4, 1}, Vec{256, 2}, Vec{1, 1}});
+    }
+    // For ApiHipRt
+    else if constexpr(std::is_same_v<
+                          alpaka::AccGpuUniformCudaHipRt<alpaka::ApiHipRt, alpaka::Dim<Acc>, alpaka::Idx<Acc>>,
+                          Acc>)
+    {
+        // Expected valid workdiv with this kernel
+        CHECK(workDiv == WorkDiv{Vec{2, 1}, Vec{512, 2}, Vec{1, 1}});
```

Comment:
IMO we should check that the blocksize is not 1024 because we do not know what the compiler is producing for different CUDA architectures.

Reply:
removed since architecture dependent, thanks.